### ISSUE TITLE
ownerInjection stopped working in tests in 3.11

### ISSUE THREAD

**Issue by mydea** [emberjs/ember.js#18259]
I have a test, with code like this:

```js
let service = UserSession.extend({
  myProp: []
}).create(this.owner.ownerInjection());
```

Which was working fine in 3.10. However, it stopped working in 3.11, with this error:

`Assertion Failed: An EmberObject based class, (unknown), was not instantiated correctly. You may have either used `new` instead of `.create()`, or not passed arguments to your call to super in the constructor: `super(...arguments)`. If you are trying to use `new`, consider using native classes without extending from EmberObject.`

I am aware that this is not an ideal pattern, and could be better solved by registering the service and looking it up from the owner. But I guess it is still a bug that this just stopped working now?

**Comment by mydea** [comment]
Note: It seems to be working fine in application code, only in tests, with `this.owner.ownerInjection()` it is failing.

**Comment by rwjblue** [comment]
@mydea - What kind of tests?

**Comment by rwjblue** [comment]
Can you spin up a demo repo so we can poke at why it might work in app code but not the test?

**Comment by mydea** [comment]
In all kinds of tests - acceptance, integration, unit.

I am on vacation starting today, I can try to put something together next week, if that helps!

**Comment by rwjblue** [comment]
Ya, sounds great! If others run into this in the meantime, please let us know and spin up a demo repo so we can try to figure out whats going on...

@mydea - Hope you have a great vacation!

**Comment by mydea** [comment]
Thanks :) So I tried to reproduce it, but it worked with a new repo. Maybe it is linked to engines (which is where I encountered this)?

**Comment by rwjblue** [comment]
Ya, possibly. I was thinking perhaps it only was an issue when using a custom resolver or something. I think we need to slowly ramp up the repro to figure out where the problem is coming from...

**Comment by mydea** [comment]
Hmm, I also tried a very basic engine reproduction but it also worked there. Not quite sure what is going on. Basically, `this.owner.ownerInjection()` is returning `undefined`. I'll try to find time to dig even deeper into this!

**Comment by rwjblue** [comment]
Fixed by #18780
